[Dist Profiling] Support shape recording for profiling collectives in gloo

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #52031 [Dist Profiling] Enable dist profiling for DDP
* #52004 [Dist Profiling] Enable profiling for gloo send/recv
* **#51822 [Dist Profiling] Support shape recording for profiling collectives in gloo**

Adds support for shape recording for profiling distributed collectives, for gloo. Added
both cpp and python tests to ensure that shapes are recorded properly.

Differential Revision: [D26291739](https://our.internmc.facebook.com/intern/diff/D26291739/)

rohan-varma committed:
[Dist Profiling] Support shape recording for profiling collectives in gloo

Adds support for shape recording for profiling distributed collectives, for gloo. Added
both cpp and python tests to ensure that shapes are recorded properly.

Differential Revision: [D26291739](https://our.internmc.facebook.com/intern/diff/D26291739/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Dist Profiling] Support shape recording for profiling collectives in gloo"


Adds support for shape recording for profiling distributed collectives, for gloo. Added
both cpp and python tests to ensure that shapes are recorded properly.

Differential Revision: [D26291739](https://our.internmc.facebook.com/intern/diff/D26291739/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Dist Profiling] Support shape recording for profiling collectives in gloo"


Adds support for shape recording for profiling distributed collectives, for gloo. Added
both cpp and python tests to ensure that shapes are recorded properly.

Differential Revision: [D26291739](https://our.internmc.facebook.com/intern/diff/D26291739/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Dist Profiling] Support shape recording for profiling collectives in gloo"


Adds support for shape recording for profiling distributed collectives, for gloo. Added
both cpp and python tests to ensure that shapes are recorded properly.

Differential Revision: [D26291739](https://our.internmc.facebook.com/intern/diff/D26291739/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Dist Profiling] Support shape recording for profiling collectives in gloo"


Adds support for shape recording for profiling distributed collectives, for gloo. Added
both cpp and python tests to ensure that shapes are recorded properly.

Differential Revision: [D26291739](https://our.internmc.facebook.com/intern/diff/D26291739/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Dist Profiling] Support shape recording for profiling collectives in gloo"


Adds support for shape recording for profiling distributed collectives, for gloo. Added
both cpp and python tests to ensure that shapes are recorded properly.

Differential Revision: [D26291739](https://our.internmc.facebook.com/intern/diff/D26291739/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Dist Profiling] Support shape recording for profiling collectives in gloo"


Adds support for shape recording for profiling distributed collectives, for gloo. Added
both cpp and python tests to ensure that shapes are recorded properly.

Differential Revision: [D26291739](https://our.internmc.facebook.com/intern/diff/D26291739/)

[ghstack-poisoned]

mrzzd reviewed (approved):
Great, thanks. Just that some tests fail.

rohan-varma committed:
Update on "[Dist Profiling] Support shape recording for profiling collectives in gloo"


Adds support for shape recording for profiling distributed collectives, for gloo. Added
both cpp and python tests to ensure that shapes are recorded properly.

Differential Revision: [D26291739](https://our.internmc.facebook.com/intern/diff/D26291739/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Dist Profiling] Support shape recording for profiling collectives in gloo"


Adds support for shape recording for profiling distributed collectives, for gloo. Added
both cpp and python tests to ensure that shapes are recorded properly.

Differential Revision: [D26291739](https://our.internmc.facebook.com/intern/diff/D26291739/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Dist Profiling] Support shape recording for profiling collectives in gloo"


Adds support for shape recording for profiling distributed collectives, for gloo. Added
both cpp and python tests to ensure that shapes are recorded properly.

Differential Revision: [D26291739](https://our.internmc.facebook.com/intern/diff/D26291739/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Dist Profiling] Support shape recording for profiling collectives in gloo"


Adds support for shape recording for profiling distributed collectives, for gloo. Added
both cpp and python tests to ensure that shapes are recorded properly.

Differential Revision: [D26291739](https://our.internmc.facebook.com/intern/diff/D26291739/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Dist Profiling] Support shape recording for profiling collectives in gloo"


Adds support for shape recording for profiling distributed collectives, for gloo. Added
both cpp and python tests to ensure that shapes are recorded properly.

Differential Revision: [D26291739](https://our.internmc.facebook.com/intern/diff/D26291739/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Dist Profiling] Support shape recording for profiling collectives in gloo"


Adds support for shape recording for profiling distributed collectives, for gloo. Added
both cpp and python tests to ensure that shapes are recorded properly.

Differential Revision: [D26291739](https://our.internmc.facebook.com/intern/diff/D26291739/)

[ghstack-poisoned]

rohan-varma commented:
CI passed, looks like its just rate limiting the completion though.
